Firebase/Auth/Source/FIRAuthDefaultUIDelegate.m

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2017 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import "FIRAuthUIDelegate.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface FIRAuthDefaultUIDelegate : NSObject <FIRAuthUIDelegate>
+/** @fn defaultUIDelegate
+    @brief Unavailable. Please use initWithViewController:
+ */
+- (instancetype)init NS_UNAVAILABLE;
+
+/** @fn initWithViewController:
+    @brief Initializes the instance with a view controller.
+    @param viewController The view controller as the presenting view controller in @c GOIUIDelegate.
+    @return The initialized instance.
+ */
+- (instancetype)initWithViewController:(UIViewController *)viewController NS_DESIGNATED_INITIALIZER;
+@end
+
+@implementation FIRAuthDefaultUIDelegate {
+  /** @var _viewController
+      @brief The presenting view controller.
+   */
+  UIViewController *_viewController;
+}
+
+- (instancetype)initWithViewController:(UIViewController *)viewController {
+  self = [super init];
+  if (self) {
+    _viewController = viewController;
+  }
+  return self;
+}
+
+id<FIRAuthUIDelegate> FIRAuthUIDelegateWithViewController(UIViewController *viewController) {
```

Comment:
I don't think we need this function. Just to call `alloc] initWith...` directly for all is internal.

Reply:
Done